launch_testing/launch_testing/__init__.py

```diff
@@ -35,6 +35,7 @@ def __init__(
         self.name = name
         self.launch_descriptor = launch_descriptor
         self.expected_lines = expected_lines
+        self.expected_output = b'\n'.join(self.expected_lines)
```

Comment:
It succeeds because I am using `re.search` instead of `re.match`.

If you run this in an interpreter:

```
import re
input = "baz 0\nfoo 1\nbar 2"
pattern = "foo \d\nbar \d"
re.search(pattern, input) is not None
```

you'll get that the last statement is true (meaning that the regex matched)

But if you use "match" instead of "search", the statement is false.